server/task/task-memory/src/main/scala/org/apache/james/task/eventsourcing/TaskAggregate.scala

```diff
@@ -36,13 +37,20 @@ class TaskAggregate private(val aggregateId: TaskAggregateId, private val histor
     case _ => throw new IllegalArgumentException("History must start with Created event")
   }
 
-  private val currentStatus: Status = history
+  private val currentDecisionProjection: DecisionProjection = history
     .getEvents
     .asScala
     .foldLeft(DecisionProjection.empty)((decision, event) => decision.update(event))
+
+  private val currentStatus: Status = currentDecisionProjection
```

Comment:
why do we need to duplicate that data locally ? Design wise, it's more relevant to store state into the DecisionProjection exclusively

Reply:
I kept the status here, because at this point in the aggregate it make sense to call `get` on the option. I would be unsafe to expose this on the decision projection